runtime/turing/src/lib.rs

```diff
@@ -1342,6 +1342,18 @@ impl_runtime_apis! {
 		}
 	}
 
+	#[cfg(feature = "try-runtime")]
+	impl frame_try_runtime::TryRuntime<Block> for Runtime {
+		fn on_runtime_upgrade() -> (Weight, Weight) {
+			log::info!("try-runtime::on_runtime_upgrade parachain-template.");
+			let weight = Executive::try_runtime_upgrade().unwrap();
```

Comment:
wait docs say Executive::try_runtime_upgrade() should only be used for testing purposes. do we have a separate implementation for Prod?

Reply:
i'm also confused what this does. it doesn't seem to be taking any specific migration. does it auto-detect migrations and run them even thought they're sitting in some folder and not mentioned here?

Reply:
woops missed a log there. Try-runtime should never ever ever be run in prod as it takes extra cpu. You just compile w/ the flag on for testing your runtimes. It will attempt to perform a runtime upgrade if applicable. These can include migrations if you code says so.

Reply:
this will make more sense when i add the new migration in the Double map pr